modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ERegex.java

```diff
@@ -68,8 +68,9 @@ void analyze(Locals locals) {
 
         try {
             Pattern.compile(pattern, flags);
-        } catch (PatternSyntaxException exception) {
-            throw createError(exception);
+        } catch (PatternSyntaxException e) {
+            throw new Location(location.getSourceName(), location.getOffset() + 1 + e.getIndex()).createError(
```

Comment:
I need the offset from the exception to get the location right. The `+1` bit is to get the leading `/`. What kind of fix were you thinking about? An override of `createError` maybe?